docs/tools/compile_pngs.sh

```diff
@@ -5,11 +5,19 @@
 DISPLAYNUM=5
 SCREENSHOTS=`dirname $0`/screenshots.sh
 
+# TODO: move this to a function which can be easily excluded and documented
 XVFB=$(which Xvfb)
 if [ -n $XVFB ]; then
 	Xvfb :$DISPLAYNUM -screen 0 1024x768x24 &
 	XVFBPID=$!
-	DISPLAY=:$DISPLAYNUM # this still doesn't work
+
+	# export DISPLAY=:$DISPLAYNUM
+	# Uncomment this if you want hide the spawned urxvt windows and only want to
+	# see the results in the pngs.
+	# It will display many warning messages in xdotool. In particular, the messages are:
+	#    XGetInputFocus returned the focused window of 1. This is likely a bug in the X server.
+	# but it works.
+
```

Comment:
Sorry but no, this specific change will break building docs on CI.

We do need to explicitly point urxvt & xdotool to X11-connect to `DISPLAY=:5`. This is why we're running a "virtual X-server" on address `:5` hosted by `Xvfb` in the first place.

On a desktop linux, with your change applied, the script will inherit `DISPLAY=:0` most likely — so `urxvt` will start connecting to your graphical session, you'll see the terminal — and the `Xvfb` instance will be completely unused. Xdotool will also be able to grab screenshots. So what's the problem?

The problem is, in CI containers there's no X server and no `DISPLAY=:0`. There we need Xvfb. And to not create extra discrepancies in testing, it's best to also use Xvfb locally.

You can *comment out* the `DISPLAY=:$DISPLAYNUM` line in your local copy — but don't commit that. It will break CI.